src/gc/collector.cpp

```diff
@@ -115,8 +140,12 @@ class TraceStack {
             pop_chunk();
             assert(cur == end);
             return *--cur; // no need for any bounds checks here since we're guaranteed we're CHUNK_SIZE from the start
+        } else {
+            // We emptied the stack, but we should prepare a new chunk in case another item
```

Comment:
Ah yes.  The TraceStack was very much operating under the assumption that after the final pop it was done.